### ISSUE TITLE
Call to a member function getTranslatedAttribute() on null

### ISSUE THREAD

**Issue by PrimoRossi** [thedevdojo/voyager#4406]
## Version information

 - Laravel: v5.8.*
 - Voyager: v^1.2
 - PHP: ^7.1.3
 - Database: [MySQL 8.0)

## Description

After correcting a known BUG for the admin menu, there´s two pages witch shows this error:

`Call to a member function getTranslatedAttribute() on null`

## Expected behavior
An admin page should render

## Screenshots
```
/var/www/html/primorossi/primorossi/vendor/tcg/voyager/src/Http/Controllers/VoyagerBaseController.php
    {
        // GET THE SLUG, ex. 'posts', 'pages', etc.
        $slug = $this->getSlug($request);
 
        // GET THE DataType based on the slug
        $dataType = Voyager::model('DataType')->where('slug', '=', $slug)->first();
 
        // Check permission
        $this->authorize('browse', app($dataType->model_name));
 
        $getter = $dataType->server_side ? 'paginate' : 'get';
 
        $search = (object) ['value' => $request->get('s'), 'key' => $request->get('key'), 'filter' => $request->get('filter')];
 
        $searchNames = [];
        if ($dataType->server_side) {
            $searchable = array_keys(SchemaManager::describeTable(app($dataType->model_name)->getTable())->toArray());
            $dataRow = Voyager::model('DataRow')->whereDataTypeId($dataType->id)->get();
            foreach ($searchable as $key => $value) {
                $displayName = $dataRow->where('field', $value)->first()->getTranslatedAttribute('display_name');
                $searchNames[$value] = $displayName ?: ucwords(str_replace('_', ' ', $value));
            }
        }
 ```

## Additional context
Can´t figure it out...

**Comment by guilhermebillig** [comment]
Hi, I had the same problem, to correct I went to the bread and rewrote one of the translations I had done earlier, so it seems to me that the problem is that when translating the fields in the bread there can be a mismatch between the amount of translations and quantity of fields in the table.

**Comment by reficul** [comment]
I had the same problem too. 
Happened when I delete some fields from a table (with bread).

I had to manually remove some entries from data_rows and data_type tables to fix the problem.

**Comment by marcellopato** [comment]
Ok, but this does not solve the problem. My database has a lot of tables after all. Imagin searching wich was the last one I´ve changed or translated.

Still can´t figure it out...

**Comment by sir-octopus** [comment]
I encounter this issue in case-sensitive databases. For example, if I have a column platformContentId:
On line 50:
`$searchable = array_keys(SchemaManager::describeTable(app($dataType->model_name)->getTable())->toArray());`
SchemaManager::describeTable() returns a Collection. Each of its items use array keys which are the lowered column name.
![image](https://user-images.githubusercontent.com/25412743/65321759-346b8980-dbe4-11e9-96a9-ace4e2511054.png)
As $searchable is just the array_keys, which do not actually match the column's 'name', line 53 fails as it attempts to run first() on a query which is searching for 'platformcontentid' instead of 'platformContentId'.

This can be easily solved by changing toArray() to pluck('name') in line 50. Existing:
`$searchable = array_keys(SchemaManager::describeTable(app($dataType->model_name)->getTable())->toArray());
`
Proposed:
`$searchable = SchemaManager::describeTable(app($dataType->model_name)->getTable())->pluck('name');`

Your problem may have a different cause but I hope this helps.

**Comment by MrCrayon** [comment]
@marcellopato go to Tools->BREAD edit and save the BREAD that had fields removed in db, that will remove all data_rows that are not used anymore for that BREAD.

**Comment by MrCrayon** [comment]
@PrimoRossi
> After correcting a known BUG for the admin menu, there´s two pages witch shows this error:

Which pages?

I think that, other than what @voxlunch said for columns that are not lower case, this problem happens when you add a field in a table and then don't update BREAD configuration.
In that case the needed data_row is not created and $dataRow->where('field', $value)->first() returns null.

**Manual solution**
After adding, deleting or renaming a field in a table managed with BREAD go save again BREAD settings.

**Comment by oriolcabau** [comment]
This happens to me when i select the server side pagination in my BREAD configuration (which i need because i have a lot of records), if i disable server side pagination, all works fine. I tried saving and deleting / recreating BREAD but nothing change, the error still persist. Any solutions?

Edit: I don't have multilingual enabled on my project.

Laravel: 5.8.*
Voyager: 1.2
PHP: 7.2.10

**Comment by MrCrayon** [comment]
@oriolcabau that's because the error happens when it tries to get translated labels for search that is only available when server side mode is enabled.

If you don't have a field with uppercase letters you should be able to solve reading my previous comments.

**Comment by marcellopato** [comment]
It happens when server-side pagination in on. That´s a fact.
Turn it off and the bug disappears.

**Comment by damms005** [comment]
This bug happens at `vendor/doctrine/dbal/lib/Doctrine/DBAL/Schema/AbstractSchemaManager.php` line `811`.

The line referenced above breaks Voyager bread when column name contains uppercase characters

**Comment by MrCrayon** [comment]
@damms005 that only lower case keys not column names.
The problem with uppercase columns is explained and fixed by @voxlunch

**Comment by emptynick** [comment]
Fixed in https://github.com/the-control-group/voyager/pull/4442 (to be released in 1.3)

**Comment by Sanswan** [comment]
For those that don't understand, when you have the BREAD in Voyager and your Mysql database! check if the order is the same. like a example: Id - form_id - data - mailto - ip_address. if this has a different order is mysql! like id - data - form_id ........ you get that error.
